python_src/src/lamp_py/tableau/jobs/gtfs_rail.py

```diff
@@ -0,0 +1,400 @@
+import os
+
+import pyarrow
+import pyarrow.parquet as pq
+import sqlalchemy as sa
+
+from lamp_py.tableau.hyper import HyperJob
+
+
+class HyperGTFS(HyperJob):
+    """Base Class for GTFS Hyper Jobs"""
+
+    def __init__(
+        self,
+        gtfs_table_name: str,
+        table_query: str,
+    ) -> None:
+        HyperJob.__init__(
+            self,
+            hyper_file_name=f"LAMP_GTFS_Rail_{gtfs_table_name}.hyper",
+            remote_parquet_path=f"s3://{os.getenv('PUBLIC_BUCKET')}/lamp/tableau/rail/LAMP_GTFS_Rail_{gtfs_table_name}.parquet",
+        )
+        self.gtfs_table_name = gtfs_table_name
+        self.create_query = table_query % ""
+        self.update_query = table_query
+
+    @property
+    def parquet_schema(self) -> pyarrow.schema:
+        """Define GTFS Table Schema"""
+
+    def create_parquet(self) -> None:
+        if os.path.exists(self.local_parquet_path):
+            os.remove(self.local_parquet_path)
+
+        pq.write_table(
+            pyarrow.Table.from_pylist(
+                mapping=self.db_manager.select_as_list(
+                    sa.text(self.create_query)
+                ),
+                schema=self.parquet_schema,
+            ),
+            self.local_parquet_path,
+        )
+
+    def update_parquet(self) -> bool:
+        self.download_parquet()
+
+        max_stats = self.max_stats_of_parquet()
+
+        max_parquet_key = max_stats["static_version_key"]
+
+        max_key_query = (
+            f"SELECT MAX(static_version_key) FROM {self.gtfs_table_name};"
+        )
+
+        max_db_key = self.db_manager.select_as_list(sa.text(max_key_query))[0][
+            "max"
+        ]
+
+        # no update needed
+        if max_db_key <= max_parquet_key:
+            return False
+
+        # add WHERE clause to UPDATE query
+        update_query = self.update_query % (
+            f" WHERE static_version_key > {max_parquet_key} ",
+        )
+
+        pq.write_table(
+            pyarrow.concat_tables(
+                [
+                    pq.read_table(self.local_parquet_path),
+                    pyarrow.Table.from_pylist(
+                        mapping=self.db_manager.select_as_list(
+                            sa.text(update_query)
+                        ),
+                        schema=self.parquet_schema,
+                    ),
+                ]
+            ),
+            self.local_parquet_path,
+        )
```

Comment:
If `static_version_key` values exist in the RDS that are larger than the values in the parquet file, pull all records with larger `static_version_key` from RDS and append to existing parquet file. 

Will have to keep an eye on resource usage of performance manager instance during these operations